examples/test_util/ESmry.cpp

```diff
@@ -0,0 +1,436 @@
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+
+#include <string>
+#include <string.h>
+#include <sstream>
+#include <iterator>
+#include <iomanip>
+#include <algorithm>
+#include <unistd.h>
+#include <limits>
+#include <set>
+
+#include "EclFile.hpp"
+#include "ESmry.hpp"
+
+/*
+
+     KEYWORDS       WGNAMES        NUMS              |   PARAM index   Corresponding ERT key
+     ------------------------------------------------+--------------------------------------------------
+     WGOR           OP_1           0                 |        0        WGOR:OP_1
+     FOPT           +-+-+-+-       0                 |        1        FOPT
+     WWCT           OP_1           0                 |        2        WWCT:OP_1
+     WIR            OP_1           0                 |        3        WIR:OP_1
+     WGOR           WI_1           0                 |        4        WWCT:OP_1
+     WWCT           W1_1           0                 |        5        WWCT:WI_1
+     BPR            +-+-+-         12675             |        6        BPR:12675, BPR:i,j,k
+     RPR            +-+-+-         1                 |        7        RPR:1
+     FOPT           +-+-+-         0                 |        8        FOPT
+     GGPR           NORTH          0                 |        9        GGPR:NORTH
+     COPR           OP_1           5628              |       10        COPR:OP_1:56286, COPR:OP_1:i,j,k
+     RXF            +-+-+-         32768*R1(R2 + 10) |       11        RXF:2-3
+     SOFX           OP_1           12675             |       12        SOFX:OP_1:12675, SOFX:OP_1:i,j,jk
+
+ */
+
+ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
+{
+    std::string rootN;
+    std::vector<int> actInd;
+    bool formatted=false;
+    
+    char buff[1000];
+    getcwd( buff, 1000 );
+
+    std::string currentWorkingDir(buff);
+    std::string currentDir=currentWorkingDir;
+
+    std::string smspec_filen;
+    std::string unsmry_filen;
+
+    if (filename.substr(filename.length() - 7, 7) == ".SMSPEC") {
+        rootN = filename.substr(0,filename.length() -7);
+    } else if (filename.substr(filename.length() -8, 8) == ".FSMSPEC") {
+        rootN=filename.substr(0,filename.length() -8);
+        formatted = true;
+    } else {
+        rootN = filename;
+    }
+
+    path = currentWorkingDir;
+    updatePathAndRootName(path, rootN);
+
+    if (formatted) {
+        smspec_filen = path + "/" + rootN + ".FSMSPEC";
+        unsmry_filen = path + "/" + rootN + ".FUNSMRY";
+    } else {
+        smspec_filen = path + "/" + rootN + ".SMSPEC";
+        unsmry_filen = path + "/" + rootN + ".UNSMRY";
+    }
+
+    std::vector<std::pair<std::string,int>> smryArray;
+
+    EclFile smspec1(smspec_filen);
```

Comment:
This way - i.e. *using* a `EclFile` instance is The Right&trade; way to do it!

Reply:
I have fixed the EGrid class. ERft and ERst need access to eclfile internals and i think inheritance is better than public accessors in EclFile.